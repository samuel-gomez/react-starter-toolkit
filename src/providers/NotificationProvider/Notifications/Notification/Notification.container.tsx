import Notification, { EType, TNotification } from './Notification';

type TNotificationContainer = TNotification & {
  NotificationCmpt?: typeof Notification;
};

const NotificationContainer = ({ classModifier = '', type = EType.error, NotificationCmpt = Notification, ...rest }: TNotificationContainer) => {
  const newClassModifier = [type, classModifier].join(' ').trim();
  return <NotificationCmpt {...rest} type={type} classModifier={newClassModifier} />;
};

export default NotificationContainer;
