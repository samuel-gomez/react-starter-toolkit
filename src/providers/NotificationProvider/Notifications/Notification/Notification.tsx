import Alert from '@axa-fr/react-toolkit-alert';
import { ALERT_ICON } from '../constants';

export enum EType {
  success = 'success',
  error = 'error',
  danger = 'danger',
}

export type TNotification = {
  id: string;
  label: string;
  onClose?: () => void;
  detail?: string;
  classModifier?: string;
  type?: keyof typeof EType;
};

const Notification = ({ id, label, onClose, detail = '', classModifier = '', type = EType.error }: TNotification) => (
  <Alert key={id} icon={ALERT_ICON[type]} title={label} classModifier={`notification ${classModifier}`} onClose={onClose}>
    {detail && <p>{detail}</p>}
  </Alert>
);

export default Notification;
