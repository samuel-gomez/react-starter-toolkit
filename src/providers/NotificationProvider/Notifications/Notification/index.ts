export { default } from './Notification.container';
export { EType } from './Notification';
export type { TNotification } from './Notification';
