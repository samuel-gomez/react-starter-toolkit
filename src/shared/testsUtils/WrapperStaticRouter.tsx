/* eslint-disable react/display-name */
import { ReactNode } from 'react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

const WrapperStaticRouter =
  (route = '/') =>
  ({ children }: { children: ReactNode }) =>
    <MemoryRouterProvider url={route}>{children}</MemoryRouterProvider>;

export default WrapperStaticRouter;
