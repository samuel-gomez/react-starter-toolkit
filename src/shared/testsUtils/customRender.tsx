/* eslint-disable react/display-name */
import { render, RenderOptions } from '@testing-library/react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

import FetchProvider from 'providers/FetchProvider';
import UserProvider from 'providers/UserProvider';
import { ReactElement, ReactNode } from 'react';
import { MOCK_API_URL } from '.';

type TMockProvider = {
  [x: string]: Record<string, unknown | number | string> | string;
};

const MockProviders =
  ({ role = '', ...testMock }: TMockProvider) =>
  ({ children }: { children: ReactNode }) => {
    const useOidcAccessTokenFn = jest.fn().mockReturnValue({ accessToken: 'accessToken' });
    const useOidcUserFn = jest.fn().mockReturnValueOnce({ oidcUser: { member_of: [`CN=${role}`], name: 'Samuel Gomez' } });
    return (
      <UserProvider useOidcUserFn={useOidcUserFn}>
        <FetchProvider
          apiUrl={MOCK_API_URL}
          fetchConfig={{
            headers: { testMock: JSON.stringify(testMock) },
          }}
          useOidcAccessTokenFn={useOidcAccessTokenFn}
        >
          <MemoryRouterProvider url="">{children}</MemoryRouterProvider>
        </FetchProvider>
      </UserProvider>
    );
  };

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>, testMock: TMockProvider = { role: '' }) =>
  render(ui, { wrapper: MockProviders(testMock), ...options });

// re-export everything
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// override render method
export { customRender as render };
