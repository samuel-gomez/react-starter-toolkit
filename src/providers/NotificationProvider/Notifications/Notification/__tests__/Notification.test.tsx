import { render } from '@testing-library/react';
import { emptyFunction } from 'shared/testsUtils';
import Notification, { EType } from '../Notification';

describe('Notifications', () => {
  const defaultProps = {
    id: 'id',
    label: 'title',
    onClose: emptyFunction,
  };
  it('Render <Notification/> with required props', () => {
    const { asFragment } = render(<Notification {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render <Notification/> with all props', () => {
    const allProps = {
      ...defaultProps,
      detail: 'detail custom',
      classModifier: 'myclass',
      type: EType.success,
    };
    const { asFragment } = render(<Notification {...allProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
