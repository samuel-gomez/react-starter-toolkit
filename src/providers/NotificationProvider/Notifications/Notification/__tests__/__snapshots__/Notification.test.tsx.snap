// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Render <Notification/> with all props 1`] = `
<DocumentFragment>
  <div
    class="af-alert af-alert--notification af-alert--myclass"
    role="alert"
  >
    <div
      class="af-alert__title"
    >
      <div
        class="af-alert__title-icon"
      >
        <i
          class="glyphicon glyphicon-ok"
        />
      </div>
      <div
        class="af-alert__title-text"
      >
        title
      </div>
      <button
        class="af-alert__title-icon-close"
        type="button"
      >
        <i
          class="glyphicon glyphicon-close"
        />
      </button>
    </div>
    <div
      class="af-alert__content"
    >
      <div
        class="af-alert__content__left"
      />
      <div
        class="af-alert__content__right"
      >
        <p>
          detail custom
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Notifications Render <Notification/> with required props 1`] = `
<DocumentFragment>
  <div
    class="af-alert af-alert--notification"
    role="alert"
  >
    <div
      class="af-alert__title"
    >
      <div
        class="af-alert__title-icon"
      >
        <i
          class="glyphicon glyphicon-minus-sign"
        />
      </div>
      <div
        class="af-alert__title-text"
      >
        title
      </div>
      <button
        class="af-alert__title-icon-close"
        type="button"
      >
        <i
          class="glyphicon glyphicon-close"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
